### PR TITLE
Ticket 1117 - Auto detect compiler startup

### PR DESCRIPTION
**Pull request by acotty**
This is for SF ticket 1117

### CODE REVIEW

**Comment by acotty** [comment]
This PR in theory should also fix ticket 808 as unless the end user is too smart for their own good should not be able to select an invalid compiler (unless they select no compiler).

**Comment by obfuscated** [comment]
> 4) Remove the ability to close the dialog by the 'X' on the top right of the dialog. This ensures that the only way to continue is when a detected compiler has been selected.

Please read carefully: You **MUST NOT** prevent the user from starting the software. This is unacceptable. If he/she wants to use an invalid compiler you must not prevent this. Your only goal is to make it harder for them to do it accidentally!

After this one covered some major points:
1. This review is the opposite of simple. Why on earth there is a checkbox for the show-path - just always show it. Why is there a radio button to show all-or-detect - just show only detected. Simplicity please!
2. No hungarian notation please, the only exceptions are m_ and probably m_p (I don't personally use it). Read the style guide for details. https://wiki.codeblocks.org/index.php/Coding_style 
3. No underscores in names
4. No auto. It is personal opinion, so auto doesn't pass my reviews.

File: src/sdk/workspaceloader.cpp [hunk 1]
```diff
@@ -160,10 +167,23 @@ bool WorkspaceLoader::Open(const wxString& filename, wxString& Title)
...
-        cbMessageBox(wxString::Format(_("%d projects could not be loaded.\nPlease see the Log window for details"), failedProjects),
-                     _("Opening WorkSpace"), wxICON_WARNING);
+    if (iFailedProjects > 0)
+    {
+        wxString sMessage = wxString::Format(_("%d projects could not be loaded.\nPlease see the Log window for details.\nThe failed projects are: %s"), iFailedProjects, sFailedProjects);
+        if (    (iFailedProjects == 1) &&
+                sFailedProjects.Contains("FortranProject") &&
+                (
+                    filename.Contains("CodeBlocks_wx31_64.workspace") ||
+                    filename.Contains("CodeBlocks_wx31.workspace")
```
Comment (obfuscated) [inline src/sdk/workspaceloader.cpp:177]: So the only purpose of codeblocks is to develop codeblocks on Windows. Please remove this.

File: src/sdk/workspaceloader.cpp [hunk 2]
```diff
@@ -96,7 +96,8 @@ bool WorkspaceLoader::Open(const wxString& filename, wxString& Title)
         return false;
     }
 
-    int failedProjects = 0;
+    int iFailedProjects = 0;
+    wxString sFailedProjects =  wxEmptyString;
```
Comment (obfuscated) [inline src/sdk/workspaceloader.cpp:100]: Problems here:
1. hungarian notation actually making code less readable, because there are now two variables with the almost same names.
2. wxEmptyString
3. Why is this change necessary? What does it fix. To me it seems that it has to be removed.

File: src/sdk/resources/auto_detect_compilers.xrc [hunk 3]
```diff
@@ -65,14 +95,12 @@
 				<border>8</border>
 			</object>
 			<object class="sizeritem">
-				<object class="wxStdDialogButtonSizer">
-					<object class="button">
-						<object class="wxButton" name="wxID_OK">
-							<label></label>
-						</object>
+				<object class="wxBoxSizer">
```
Comment (obfuscated) [inline src/sdk/resources/auto_detect_compilers.xrc:98]: Removing the std-button-sizer is unacceptable it was added on purpose. It should span the whole bottom row of the dialog.

File: src/sdk/resources/auto_detect_compilers.xrc [hunk 4]
```diff
@@ -22,16 +22,46 @@
...
+								<flag>wxALL|wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+								<border>5</border>
+							</object>
+							<object class="sizeritem">
+								<object class="wxCheckBox" name="cbShowCompilerPath">
+									<label>Show Path</label>
+								</object>
+								<flag>wxALL|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL</flag>
+								<border>5</border>
+							</object>
```
Comment (obfuscated) [inline src/sdk/resources/auto_detect_compilers.xrc:60]: Useless complexity. Please remove.

File: src/sdk/resources/auto_detect_compilers.xrc [hunk 5]
```diff
@@ -3,12 +3,12 @@
...
 		<title>Compilers auto-detection</title>
 		<centered>1</centered>
-		<style>wxDEFAULT_DIALOG_STYLE|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
+		<style>wxCAPTION|wxRESIZE_BORDER|wxMAXIMIZE_BOX</style>
 		<object class="wxBoxSizer">
 			<orient>wxVERTICAL</orient>
 			<object class="sizeritem">
 				<object class="wxStaticText" name="ID_STATICTEXT2">
-					<label>Note: After auto-detection, at least one compiler&apos;s master path is still empty and therefore invalid.&#x0A;Inspect the list below and change the compiler&apos;s master path later in the compiler options.&#x0A;Select you favourite default compiler here:</label>
+					<label>Note: After auto-detection, either the default compiler was not detected or at least none compiler&apos;s master&#x0A;path is still empty and therefore invalid. Inspect the list below and select another default compiler if it &#x0A;was not found or and change the compiler&apos;s master path later in the compiler options if the compiler is invalid.&#x0A;Select you favourite default compiler here:</label>
```
Comment (obfuscated) [inline src/sdk/resources/auto_detect_compilers.xrc:11]: `found or and change` - what is the meaning of this?

File: src/sdk/compilerfactory.cpp [hunk 6]
```diff
@@ -277,9 +279,15 @@ void CompilerFactory::LoadSettings()
...
-            Manager::Get()->GetLogManager()->DebugLog(F(_T("Master path of compiler ID \"%s\" is empty -> triggers auto-detection."), Compilers[i]->GetID().wx_str()));
+        if (    (Compilers[i]->GetMasterPath().IsEmpty() || !wxFileName::DirExists(Compilers[i]->GetMasterPath()))
+                &&
+                Compilers[i]->GetID().IsSameAs(defaultCompilerID)
+           )
+        {
+            if (Compilers[i]->GetMasterPath().IsEmpty())
+                Manager::Get()->GetLogManager()->DebugLog(F(_T("The master path of compiler ID \"%s\" is empty -> triggers auto-detection."), Compilers[i]->GetID().wx_str()));
+            if (!wxFileName::DirExists(Compilers[i]->GetMasterPath()))
+                Manager::Get()->GetLogManager()->DebugLog(F(_T("The master path (%s) of compiler ID \"%s\" does not exist -> triggers auto-detection."), Compilers[i]->GetMasterPath(), Compilers[i]->GetID().wx_str()));
```
Comment (obfuscated) [inline src/sdk/compilerfactory.cpp:290]: 1. One common thing people like to do unfortunately - copy-paste repeated getters. Please store Compilers[i]->GetMasterPath() and Compilers[i]->GetID() in variables. Even repeating Compilers[i] on every line is a waste.
2. Please do not use F, use wxString::Format please.
3. Do not use wx_str() in format. Passing just wxStrings work.
4. Translate the messages you print in the log
5. Do not use _T or wxT. They are useless in wx3.x.

File: src/sdk/autodetectcompilers.cpp [hunk 7]
```diff
@@ -180,7 +327,18 @@ void AutoDetectCompilers::OnUpdateUI(wxUpdateUIEvent& event)
 {
     wxListCtrl* list = XRCCTRL(*this, "lcCompilers", wxListCtrl);
     bool en = list->GetNextItem(-1, wxLIST_NEXT_ALL, wxLIST_STATE_SELECTED) != -1;
-    XRCCTRL(*this, "btnDefault", wxButton)->Enable(en);
+    XRCCTRL(*this, "btnSetDefaultCompiler", wxButton)->Enable(en);
 
+    wxString sDefaultCompiler = CompilerFactory::GetDefaultCompiler()->GetName();
+    for (auto itCL = std::begin(vCompilerList); itCL != std::end(vCompilerList); ++itCL)
```
Comment (obfuscated) [inline src/sdk/autodetectcompilers.cpp:333]: Range based for loop (without the auto) would be a lot better here.

File: src/sdk/autodetectcompilers.cpp [hunk 8]
```diff
@@ -40,109 +45,240 @@ AutoDetectCompilers::AutoDetectCompilers(wxWindow* parent)
...
+
+            int idx = list->GetItemCount();
+            list->InsertItem(idx, itCL->wxsCompilerName);
+            list->SetItem(idx, ccnStatusColumn, itCL->wxsStatus);
+            if (bShowCompilerPath)
+                list->SetItem(idx, ccnDetectedPathColumn, itCL->wxsCompilerPath);
+
+            switch(itCL->eHighlight)
+            {
+            case CompilerHighlightColor::eHighlightGrey:
```
Comment (obfuscated) [inline src/sdk/autodetectcompilers.cpp:259]: Formatting is broken here.

File: src/sdk/autodetectcompilers.cpp [hunk 9]
```diff
@@ -40,109 +45,240 @@ AutoDetectCompilers::AutoDetectCompilers(wxWindow* parent)
...
+                case CompilerHighlightColor::eHighlightGreen:
+                    list->SetItemBackgroundColour(idx, *wxGREEN);
+                    break;
+                case CompilerHighlightColor::eHighlightRed:
+                    list->SetItemBackgroundColour(idx, *wxRED);
+                    break;
+                case CompilerHighlightColor::eHighlightYellow:
+                    list->SetItemBackgroundColour(idx, *wxYELLOW);
+                    break;
+                case CompilerHighlightColor::eHighlightNone:
```
Comment (obfuscated) [inline src/sdk/autodetectcompilers.cpp:271]: Typical offence - set the BG color without setting the FG colour or vice-versa. Hardcoding colours in this dialog is wrong idea. It doesn't probably work well with dark-themes or red-themes or green-themes. There are strange people around the world, please don't make them suffer even more.

I think it is best to remove any colouring from this dialog. Colours could be used in other places, but the cbColourManager should be available. Here it is not.

File: src/sdk/autodetectcompilers.cpp [hunk 10]
```diff
@@ -40,109 +45,240 @@ AutoDetectCompilers::AutoDetectCompilers(wxWindow* parent)
...
-        CompilerFactory::SetDefaultCompiler(idx);
-        XRCCTRL(*this, "lblDefCompiler", wxStaticText)->SetLabel(CompilerFactory::GetDefaultCompiler()->GetName());
+        list->ClearAll();
+        list->InsertColumn(ccnNameColumn, _("Compiler"), wxLIST_FORMAT_LEFT, 380);
+        list->InsertColumn(ccnStatusColumn, _("Status"), wxLIST_FORMAT_LEFT, 100);
+
+        if (bShowCompilerPath)
+            list->InsertColumn(ccnDetectedPathColumn, _("Compiler Path"), wxLIST_FORMAT_LEFT, 200);
+
+        for (auto itCL = std::begin(vCompilerList); itCL != std::end(vCompilerList); ++itCL)
```
Comment (obfuscated) [inline src/sdk/autodetectcompilers.cpp:246]: One more auto...

File: src/include/autodetectcompilers.h [hunk 11]
```diff
@@ -13,11 +13,41 @@ class wxUpdateUIEvent;
...
+
+        struct CompilerItem
+        {
+            wxString wxsCompilerName;
+            wxString wxsStatus;
+            wxString wxsCompilerPath;
+            CompilerHighlightColor eHighlight;
+            bool bDetected;
+        };
+        std::vector<CompilerItem> vCompilerList;
```
Comment (obfuscated) [inline src/include/autodetectcompilers.h:41]: Indent here is a bit odd, please fix.

Comment (obfuscated) [reply src/include/autodetectcompilers.h:41]: vCompilerList doesn't follow the convention for members.
I prefer to put members at the bottom of the class.

Using a vector requires an include. This would fail no pch builds.

File: src/sdk/autodetectcompilers.cpp [hunk 12]
```diff
@@ -40,109 +45,240 @@ AutoDetectCompilers::AutoDetectCompilers(wxWindow* parent)
...
+                {
+                    wxColour  colorDefault =  XRCCTRL(*this, "ID_STATICTEXT1", wxStaticText)->GetBackgroundColour();
+
+                    controlLblDefCompiler->SetLabel(cItem.wxsCompilerName);
+                    controlLblDefCompiler->SetBackgroundColour(colorDefault);
+                }
+                else
+                {
+                    controlLblDefCompiler->SetLabel(wxString::Format(_("INVALID: %s"),cItem.wxsCompilerName));
+                    controlLblDefCompiler->SetBackgroundColour(*wxRED);
```
Comment (obfuscated) [inline src/sdk/autodetectcompilers.cpp:192]: This is unacceptable. Changing just the background breaks on non-default themes. Please remove this colour changing.

**Review by obfuscated — COMMENTED** [review]
First pass on the code done. A bit backwards, so sorry about that. :)

**Comment by acotty** [comment]
Thanks for taking the time to look at the change. I am working through the issues you have raised and working on resolving them or in some cases I will need to ask for more details/lionks to how this can be resolved or supply additional details  and therefore askwhat to do next based on the extra info. Do you want me to resolve one issue at a time or all at once?

I am asking again as a number of the issues are with the coding standard, is there a current documented coding standard anywhere that I need to adhere to as I will need to make changes to the other PR's to ensure that I meet "it". If there is not one is it possible for you to create one so I can ensure that I can adhere to it. I know it will evolve and the first n versions will not include all of the requirements , but at least it will be a start.

File: src/sdk/workspaceloader.cpp (hunk 2 above)
Comment (acotty) [reply src/sdk/workspaceloader.cpp:100]: 1 & 2 okay. 
3. If you are building from a batch file then at this point in time there is no log shown and as such showing the first X (i decided 10) failed projects made it so that you could continue the batch build or not. In the batch build the dialog if any project failed is always shown.
So based on this do you still want it removed?

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 3 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc:98]: See ticket 1110, when loading xrc in wxSmith wsWidget throws a debug alert with the wxStdDialogButtonSizer and this fixed the alert. So do you still want me to revert the change?

The debug alerts details are:
[Window Title]
wxWidgets Debug Alert

[Main Instruction]
A debugging check in this application has failed.

[Content]
../../src/common/wincmn.cpp(2490): assert ""!m_containingSizer"" failed in SetContainingSizer(): Adding a window already in a sizer, detach it first!

[ ] Don't show this dialog again  [Stop] [Continue]

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 4 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc:60]: This is needed if the if the wxStdDialogButtonSizer change stays otherwise the OK button moves to the very left.

File: src/sdk/workspaceloader.cpp (hunk 1 above)
Comment (acotty) [reply src/sdk/workspaceloader.cpp:177]: I put this in as I do not have the fortran source code, so every time I do a batch build I do not want the pop up to indicate the Fortran project could not be loaded.

I will remove it.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 5 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc:11]: Looks like I did not save the update I did....

File: src/sdk/autodetectcompilers.cpp (hunk 9 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp:271]: 100% agree with this as a typical offence when I searched for ways to fix it.

How in this area how do I make it easier for the user to see which compiler has a good or bad path or the autodetect code has detected the path has changed?

File: src/sdk/workspaceloader.cpp (hunk 2 above)
Comment (obfuscated) [reply src/sdk/workspaceloader.cpp:100]: Yes, it is totally unrelated.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 4 above)
Comment (obfuscated) [reply src/sdk/resources/auto_detect_compilers.xrc:60]: This comment is related to the radiobox and the checkbox, not about the alignment.
I want no checkbox and no radiobox.

File: src/sdk/autodetectcompilers.cpp (hunk 9 above)
Comment (obfuscated) [reply src/sdk/autodetectcompilers.cpp:271]: > How in this area how do I make it easier for the user to see ...

Something that is reliable in a themeable world.

>...  or the autodetect code has detected the path has changed?

Why would you want to show this?
When does it happen?

File: src/sdk/workspaceloader.cpp (hunk 2 above)
Comment (acotty) [reply src/sdk/workspaceloader.cpp:100]: This has been removed in the next code push for this PR. I have created ticket 1128 for this and it will be included in the OR for this ticket.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 3 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc:98]: Who is the wxwidget XRC / wxsmith expert who can help with resolving this as I do not want to revert it and cause people headaches with the assert occurring.

File: src/sdk/autodetectcompilers.cpp (hunk 9 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp:271]: > > How in this area how do I make it easier for the user to see ...
>
> Something that is reliable in a themeable world.

So what is reliable in the themeable world with the current C::B code? This dialog only pops up on first install or if something changes with the default compiler on subsequent startups. For the vast majority of users it will only pop up once on the initial install. I am the exception at the moment due to the testing and swapping in/out different compilers for testing purposes in order to text the various combinations and the pop up itself.


> > ...  or the autodetect code has detected the path has changed?
>
> Why would you want to show this?

I want to show this to confirm to the end user that the compiler directory change is okay and acceptable instead of C::B  automatically changing the directory on startup without letting the end user know that it occurred.

> When does it happen?

This occurs when the end users moves or renames a compiler directory on their C: or when they switch in/out different compilers on their C: (there are so many different MingW versions out there that I expect this will occur more that the move/rename).

File: src/sdk/autodetectcompilers.cpp (hunk 10 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp:246]: Changed to the following in the next code push:
for (std::vector<CompilerItem>::iterator it = m_CompilerList.begin(); it != m_CompilerList.end(); ++it)

Is this acceptable?

**Comment by acotty** [comment]
You were 100% correct, I missed one line. I pulled down the coding standard and have added to it based on this PR and the changes you made in the following commit:
https://github.com/obfuscated/codeblocks_sf/commit/b1e0b9cd3c2abcf8c87e21c764f0a5ecab4f2e5f

File: src/include/autodetectcompilers.h [hunk 13]
```diff
@@ -1,27 +1,60 @@
...
-        DECLARE_EVENT_TABLE()
-};
-
-#endif // AUTODETECTCOMPILERS_H
+/*
+ * This file is part of the Code::Blocks IDE and licensed under the GNU Lesser General Public License, version 3
+ * http://www.gnu.org/licenses/lgpl-3.0.html
+ */
+
+#ifndef AUTODETECTCOMPILERS_H
```
Comment (obfuscated) [inline src/include/autodetectcompilers.h:6]: This whole file appears to be changed. Please fix your auto-crlf setting!

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 3 above)
Comment (obfuscated) [reply src/sdk/resources/auto_detect_compilers.xrc:91]: No idea. Probably there is none. Miguel Gimenez has made some changes to wxSmith recently.

But even if you refuse to revert this change, I'll revert it when I decide to push the changes to SVN.
The issue about the sizer is unrelated and it should be resolved separately. Doing work and then reverting it doesn't help!

File: src/sdk/autodetectcompilers.cpp [hunk 14]
```diff
@@ -12,137 +12,325 @@
...
+        }
+    }
+    return true;
+}
+
+void AutoDetectCompilers::OnClose(wxCloseEvent& event)
+{
+    if ( event.CanVeto())
+    {
+        if (!closeCheckOkay())
```
Comment (obfuscated) [inline src/sdk/autodetectcompilers.cpp:233]: I don't like this. If this review passes, I'll just remove this code.
Forcing people to select a default compiler is not a good idea.
Probably wouldn't use this compiler once.

File: src/sdk/resources/auto_detect_compilers.xrc [hunk 15]
```diff
@@ -22,16 +22,39 @@
...
+								<object class="wxButton" name="btnSetDefaultCompiler">
+									<label>Set as default</label>
+									<pos>-1,-1</pos>
+									<size>-1,-1</size>
+								</object>
+								<flag>wxALIGN_CENTER_HORIZONTAL|wxALIGN_CENTER_VERTICAL</flag>
+								<border>5</border>
+							</object>
+							<object class="sizeritem">
+								<object class="wxRadioBox" name="rbCompilerShowOptions">
```
Comment (obfuscated) [inline src/sdk/resources/auto_detect_compilers.xrc:43]: I've requested for this to be removed. Why are you refusing to do this?

File: src/include/autodetectcompilers.h [hunk 16]
```diff
@@ -17,11 +18,44 @@ class AutoDetectCompilers : public wxScrollingDialog
 		AutoDetectCompilers(wxWindow* parent);
 		~AutoDetectCompilers() override;
 
-		void OnDefaultClick(wxCommandEvent& event);
+        bool closeCheckOkay();
+        void OnClose(wxCloseEvent& event);
```
Comment (obfuscated) [inline src/include/autodetectcompilers.h:22]: The indent is wrong here...

Comment (acotty) [reply src/include/autodetectcompilers.h:22]: Can you check your editor to ensure that tabs are set to show as 4 spaces.  The lines I have added have 8 spaces and the lines above and below have 2 x tab characters.

Most lines use spaces.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 15 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc:43]: See the original post above for reason.

File: src/sdk/autodetectcompilers.cpp (hunk 14 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp:233]: They do not anymore.

File: src/include/autodetectcompilers.h (hunk 13 above)
Comment (acotty) [reply src/include/autodetectcompilers.h]: Should be fixed.  My initial new computer setup was wrong. I hope it is correct now.

File: src/sdk/autodetectcompilers.cpp (hunk 12 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp]: The existing code does this as well.

This is the existing code for lines 120 to 123.
            if (highlight == 1)
                list->SetItemBackgroundColour(idx, *wxRED);
            else if (highlight == -1)
                list->SetItemTextColour(idx, *wxLIGHT_GREY);

File: src/include/autodetectcompilers.h (hunk 11 above)
Comment (acotty) [reply src/include/autodetectcompilers.h]: Any updated on the indenting? Can you check your editor settings as it looks okay on my end.

File: src/sdk/autodetectcompilers.cpp (hunk 10 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp]: The code fix has been pushed.

File: src/sdk/autodetectcompilers.cpp (hunk 9 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp]: Any updates?

File: src/sdk/autodetectcompilers.cpp (hunk 8 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp]: I have not changed this as it is in an area of code that will need to be reworked due to the color highlighting issue.

File: src/sdk/compilerfactory.cpp (hunk 6 above)
Comment (acotty) [reply src/sdk/compilerfactory.cpp:290]: This issue was hidden and I did not see it until tonight. I will need to make code changes to resolve these issues.

File: src/sdk/autodetectcompilers.cpp (hunk 7 above)
Comment (acotty) [reply src/sdk/autodetectcompilers.cpp]: The code change to resolve this have been pushed.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 5 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc]: The code change to resolve this have been pushed.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 3 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc:91]: Due to this issue, this PR will not be able to be resolved until at least the middle of September as this is when Miguel Gimenez may have some time to look at the wxStdDialogButtonSizer issue.

File: src/sdk/workspaceloader.cpp (hunk 2 above)
Comment (acotty) [reply src/sdk/workspaceloader.cpp]: Code has been pushed to resolve 1 & 2.

File: src/sdk/workspaceloader.cpp (hunk 1 above)
Comment (acotty) [reply src/sdk/workspaceloader.cpp]: Code has been pushed to resolve this.

File: src/include/autodetectcompilers.h (hunk 16 above)
Comment (obfuscated) [reply src/include/autodetectcompilers.h:22]: I don't use an editor other than the diff view of github.

File: src/sdk/autodetectcompilers.cpp (hunk 14 above)
Comment (obfuscated) [reply src/sdk/autodetectcompilers.cpp:233]: You've committed the too many message boxes crime. Nobody reads those, especially people new to the software which just want to type some code.

File: src/sdk/autodetectcompilers.cpp (hunk 12 above)
Comment (obfuscated) [reply src/sdk/autodetectcompilers.cpp]: So the existing code is also wrong. It is not a valid excuse.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 3 above)
Comment (obfuscated) [reply src/sdk/resources/auto_detect_compilers.xrc:91]: Really, you'll block work on this due to one unrelated minor issue? Keep in mind that I'll not look at any of your other PRs until this one is resolved. I don't have the capacity at the moment.

File: src/sdk/autodetectcompilers.cpp (hunk 9 above)
Comment (obfuscated) [reply src/sdk/autodetectcompilers.cpp]: > I want to show this to confirm to the end user that the compiler directory change is okay and acceptable instead of C::B automatically changing the directory on startup without letting the end user know that it occurred.

This is wrong. C::B must not detect changes to the compiler after initial setup. If the compiler setup on the machine changes, I think people should be expected to update their settings.

> This occurs when the end users moves or renames a compiler directory on their C: or when they switch in/out different compilers on their C: (there are so many different MingW versions out there that I expect this will occur more that the move/rename).

I don't think this case should be handled at all. The auto-detect dialog should be shown only to people which don't have default.conf. Nobody expects or wants this dialog to be shown. This adds just useless and annoying complexity to the code.

File: src/sdk/resources/auto_detect_compilers.xrc (hunk 3 above)
Comment (acotty) [reply src/sdk/resources/auto_detect_compilers.xrc:91]: Yes.

**Comment by acotty** [comment]
Was informed this PR will not be looked at, so closing it.
